btk/metrics.py

```diff
@@ -1,484 +1,167 @@
-    detected_table["match_true_id1"] = np.ones(num_det, dtype=int) * -1
-    detected_table["match_true_id2"] = np.ones(num_det, dtype=int) * -1
-    detected_table["match_galtileid1"] = np.ones(num_det, dtype=int) * -1
-    detected_table["match_galtileid2"] = np.ones(num_det, dtype=int) * -1
+import skimage.metrics
 
 
 def get_detection_match(true_table, detected_table):
-    """Match detections to true objects and update values in the input
-    blend catalog and detection catalog.
-
-    Function does not return anything, only the astropy tables are updated.
+    """Match detections to true objects.
 
     Args:
         true_table (astropy.table.Table): Table with entries corresponding to
             the true object parameter values in one blend.
         detected_table(astropy.table.Table): Table with entries corresponding
             to output of measurement algorithm in one blend.
+    Returns:
+        match_table (astropy.table.Table) : Table containing the matches for the
+            true galaxies, in the same order as true_table
     """
+    match_table = astropy.table.Table()
     if len(detected_table) == 0 or len(true_table) == 0:
         # No match since either no detection or no true objects
         return
-    t_x = true_table["x_peak"][:, np.newaxis] - detected_table["x_peak"]
-    t_y = true_table["y_peak"][:, np.newaxis] - detected_table["y_peak"]
+    t_x = true_table["x_peak"] - detected_table["x_peak"][:, np.newaxis]
+    t_y = true_table["y_peak"] - detected_table["y_peak"][:, np.newaxis]
     dist = np.hypot(t_x, t_y)
-    norm_size = true_table["size"]
-    norm_dist = dist / norm_size[:, np.newaxis]
-    detected_table["dSigma_min"] = np.min(norm_dist, axis=0)
-    detected_table["d_min"] = np.min(dist, axis=0)
-    detection_threshold1 = 5
-    condlist1 = [
-        np.min(dist, axis=0) <= detection_threshold1,
-        np.min(dist, axis=0) > detection_threshold1,
+    match_table["d_min"] = np.min(dist, axis=0)
+    detection_threshold = 5
+    condlist = [
+        np.min(dist, axis=0) <= detection_threshold,
+        np.min(dist, axis=0) > detection_threshold,
     ]
-    choicelist1 = [np.argmin(dist, axis=0), -1]
-    match_id1 = np.select(condlist1, choicelist1)
-    detected_table["match_true_id1"] = match_id1
-    detected_table["match_galtileid1"] = true_table["galtileid"][match_id1]
-    detection_threshold2 = 0.5
-    condlist2 = [
-        np.min(norm_dist, axis=0) <= detection_threshold2,
-        np.min(norm_dist, axis=0) > detection_threshold2,
+    choicelist = [np.argmin(dist, axis=0), -1]
+    match_id = np.select(condlist, choicelist)
+    # Not perfect... See how to improve that
+    for m in np.unique(match_id[match_id >= 0]):
+        idx = np.where(match_id == m)[0]
+        best_match = match_table["d_min"][idx].argmin()
+        for i in range(len(idx)):
+            if i != best_match:
+                match_id[idx[i]] = -1
```

Comment:
is this block of code because sometimes you could have multiple matches in `match_id` to the same true object?

Reply:
Yeah exactly. We are matching each true object with the closest detected object, but this is not enough because we don't want to have 2 objects matched with the same detection, so we only keep the one which is the closest. This may not be the perfect way to do it since it can lead to cases were you have two detections and two objects but only one object matched...